### PR TITLE
Add "Remove jQuery" guide

### PR DESCRIPTION
**Pull request by simonihmig**
Based on [RFC 294](https://github.com/emberjs/rfcs/pull/294).

I basically just resubmitted https://github.com/emberjs/guides/pull/2191, which @jenweber already reviewed back then...

### CODE REVIEW

**Comment by simonihmig** [comment]
Hm, the failing tests seem to be unrelated! (broken link in `defining-a-component.md`?)

**Comment by jenweber** [comment]
@simonihmig the old PR was marked WIP - should this one be merged in or is more work needed?

**Comment by locks** [comment]
Weren't we going to have a top-level "Optional Features" section?

**Comment by simonihmig** [comment]
@jenweber @locks I think we should try to get this merged. Open questions:
* where to put it, under "Managing Dependencies" or some new category ("Optional Features")? If there is no consensus yet, I would suggest to leave it like it is now, and refactor later, otherwise this remains undocumented for too long IMHO.
* IIRC Ember CLI 3.3 was a republish of 3.2, so Ember CLI 3.4 is the one version that correctly implements the opt out stuff. Should be mentioned probably?
* this is still targeting the 3.3 docs, should this rather be moved to the current 3.4 docs?

**Comment by jenweber** [comment]
@locks do you want to find a home for this? 

Otherwise I would like to make it a fairly generic url (since urls are forever) and merge.

**Comment by jenweber** [comment]
Also thanks @simonihmig for your patience. We have let it go too long without a firm decision. The hesitation is that once a url is made in the guides, it becomes something we have to maintain indefinitely, or we break a whole lot of people's urls that point to `/release/`

**Comment by jenweber** [comment]
@locks do you have a decision on where this should go? I suggest we get it in within the next week or so.

This PR was still open as of creating the 3.5 release. This means before this PR to be merged, any changes requested should be made, and then the same additions should be applied to the upcoming 3.5 version files. They will be available later this week. Sorry about the extra step!

**Comment by locks** [comment]
I backported to the v3.3 guides since the optional feature was introduced in that release.

**Comment by jenweber** [comment]
Looks good to me! I would like to hear from @simonihmig before we merge. Thanks @locks!

File: guides/v3.3.0/configuring-ember/optional-features.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+As we can see from the warning, `@ember/optional-features` has created a file in `config/optional-features.json` to store the configuration for your project.
+We commit it to our repository and we are off to the races!
+
+### jquery-integration
+
+The Ember framework comes by default with jQuery integration.
+It is used for event handling, and to provide some APIs like `this.$()` in components.
+
+With the release of ember-source v3.0.0, an optional feature flag was introduced that allows users to opt out of jQuery.
```
Comment (simonihmig) [inline guides/v3.3.0/configuring-ember/optional-features.md]: Should be 3.3.0! (also requires Ember CLI 3.4.0 to work properly)

File: guides/v3.3.0/configuring-ember/optional-features.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  jquery-integration (Default: true)
+    Adds jQuery to the Ember application.
+    More information: https://github.com/emberjs/rfcs/pull/294
+```
+
+## Features
+
+Once you see a feature that you would like to toggle for your project you can run one of two commands, `ember feature:enable <feature>` and `ember feature:disable <feature>`.
+
+Let us disable `template-only-glimmer-components` to see what happens:
```
Comment (simonihmig) [inline guides/v3.3.0/configuring-ember/optional-features.md]: That feature is not listed under the `Available features` output above. Should we add that?

File: guides/v3.3.0/configuring-ember/optional-features.md (hunk 1 above)
Comment (locks) [reply guides/v3.3.0/configuring-ember/optional-features.md]: hm 🤔

**Pull request merged**